lib/license_finder/decision_applier.rb

```diff
@@ -28,10 +28,14 @@ def any_packages?
 
     def apply_decisions
       all_packages
-        .map { |package| with_decided_licenses(package) }
-        .map { |package| with_approval(package) }
-        .map { |package| with_homepage(package) }
         .reject { |package| ignored?(package) }
+        .map do |package|
+          with_homepage(
+            with_approval(
+              with_decided_licenses(package)
+            )
+          )
+        end
```

Comment:
I thought we can reduce iterations, within one of our projects we have thousands of packages and reducing number of iterations can be already recognized regarding runtime.